src/Adapter/Order/CommandHandler/AddCartRuleToOrderHandler.php

```diff
@@ -111,12 +112,21 @@ public function handle(AddCartRuleToOrderCommand $command): void
                         );
                     }
                 } else {
-                    throw new OrderException('Percentage discount value cannot be higher than 100%.');
+                    throw new NegativeCartRuleValueException(
+                        'Reduction percentage must be between 0% and 100%.',
+                        NegativeCartRuleValueException::NEGATIVE_PERCENTAGE
+                    );
                 }
 
                 break;
             // Amount type
             case OrderDiscountType::DISCOUNT_AMOUNT:
+                if (0 > $discountValue) {
```

Comment:
Same here.
From `order page` legacy, it is possible to create percentage discount with value 0%, but when visiting this cart rule later in `catalog edit cart rule` legacy, it has no percentage reduction at all. In fact you can't create percentage discount with value 0% in `catalog edit cart rule` legacy. Wdyt?